File: 2nd Sem/DS/Programs/pattern.c
#include<stdio.h>
void pattern1(int n)
{
  if (n < 1)
  return;

  printf("*");
  pattern1(n-1);
}

void pattern(int n)
{
  if(n < 1)
  return;

  pattern1(n);
  printf("\n");
  pattern(n-1);
}

int main()
{
  int n;
  printf("Enter the size:");
  scanf("%d",&n);
  pattern(n);
  return 0;
}

File: 4th Sem/Java/Programs/Week 3-practice/Pattern.java
import java.util.Scanner;
 
public class Pattern {
 
    public static void main(String[] args) {
        int num, space;
 
        System.out.print("Enter a number between 1 to 9 : ");
 
        Scanner reader = new Scanner(System.in);
        try {
            num = reader.nextInt();
 
            space = num - 1;
 
            for (int i = 1; i <= num; i++) {
                for (space = 1; space <= (num - i); space++) {
                    System.out.print(" ");
                }
 
                for (int j = 1; j <= i; j++) {
                    System.out.print(j);
                }
 
                for (int k = (i - 1); k >= 1; k--) {
                    System.out.print(k);
                }
 
                System.out.println();
 
            }
        } finally {
            reader.close();
        }
    }
}